src/sentry/utils/kvstore/bigtable.py

```diff
@@ -0,0 +1,287 @@
+                # else who already initialized, so we first check again to make
+                # sure this isn't the case.
+                try:
+                    table = self.__table
+                except AttributeError:
+                    table = self.__table = (
+                        bigtable.Client(project=self.project, **self.client_options)
+                        .instance(self.instance)
+                        .table(self.table_name)
+                    )
+            return table
+
+    def get(self, key: str) -> Optional[bytes]:
+        row = self._get_table().read_row(key)
+        if row is None:
+            return None
+
+        return self.__decode_row(row)
+
+    def get_many(self, keys: Sequence[str]) -> Iterator[Tuple[str, bytes]]:
+        rows = RowSet()
+        for key in keys:
+            rows.add_row_key(key)
+
+        for row in self._get_table().read_rows(row_set=rows):
+            value = self.__decode_row(row)
+
+            # Even though Bigtable in't going to return empty rows, an empty
+            # value may be returned by ``__decode_row`` if the the row has
+            # outlived its TTL, so we need to check its value here.
+            if value is not None:
+                yield row.row_key.decode("utf-8"), value
+
+    def __decode_row(self, row: PartialRowData) -> Optional[bytes]:
+        columns = row.cells[self.column_family]
+
+        try:
+            cell = columns[self.data_column][0]
+        except KeyError:
+            logger.warning("Retrieved row (%r) which does not contain a data column!", row.row_key)
+            return None
+
+        # Check if a TTL column exists for this row. If there is, we can use
+        # the `timestamp` property of the cells to see if we should return the
+        # row or not.
+        if self.ttl_column in columns:
+            # NOTE: If you need the value, it can be unpacked with the struct.
+            if cell.timestamp < timezone.now():
+                return None
+
+        value = cast(bytes, cell.value)
+
+        if self.flags_column in columns:
+            flags = self.Flags(self.flags_struct.unpack(columns[self.flags_column][0].value)[0])
+
+            # Check if there is a compression flag set, if so decompress the value.
+            # XXX: If no compression flags are matched, we unfortunately can't
+            # tell the difference between data written with a compression
+            # strategy that we're not aware of and data that was not compressed
+            # at all, so we just return the data and hope for the best. It is
+            # also possible that multiple compression flags match. We just stop
+            # after the first one matches. Good luck!
+            for compression_flag, strategy in self.compression_strategies.values():
+                if compression_flag in flags:
+                    value = strategy.decompress(value)
+                    break
```

Comment:
Inlined this behavior since it is shorter (it no longer behaves as a switch statement.)